src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -387,12 +387,9 @@ private boolean checkLoop(final DetailAST ast) {
     private boolean checkTry(final DetailAST ast, boolean useBreak,
                              boolean useContinue) {
         final DetailAST finalStmt = ast.getLastChild();
-        boolean isTerminated = false;
-        if (finalStmt.getType() == TokenTypes.LITERAL_FINALLY) {
```

Comment:
After changing `if` to `&&` one uncovered branch was reported by Jacoco. New test case added.